wazo_call_logd/tests/test_generator_scenarios.py

```diff
@@ -0,0 +1,424 @@
+    id: int
+    uniqueid: str
+    linkedid: str
+    eventtime: datetime
+    amaflags: int = 0
+    eventtype: str = ''
+    userdeftype: str = ''
+    cid_name: str = ''
+    cid_num: str = ''
+    cid_ani: str = ''
+    cid_rdnis: str = ''
+    cid_dnid: str = ''
+    exten: str = ''
+    context: str = ''
+    channame: str = ''
+    appname: str = ''
+    appdata: str = ''
+    accountcode: str = ''
+    peeraccount: str = ''
+    userfield: str = ''
+    peer: str = ''
+    extra: str = None
+    call_log_id: int = None
+
+    def __post_init__(self, **kwargs):
+        for field in fields(self):
+            current_value = getattr(self, field.name)
+            converter = (
+                parse_eventtime
+                if field.name == 'eventtime'
+                else get_type_hints(self.__class__)[field.name]
+            )
+            if current_value is not None:
+                try:
+                    setattr(
+                        self,
+                        field.name,
+                        converter(current_value),
+                    )
+                except Exception:
+                    logger.exception(
+                        'Error processing field %s of CEL for input %s',
+                        field.name,
+                        current_value,
+                    )
+                    raise
+
+
+def raw_cels(cel_output, row_count=None):
+    '''
+    this decorator takes the output of a psql query
+    and parses it into CEL entries that are loaded into the database
+    '''
+    cels = parse_raw_cels(cel_output)
+    if row_count:
+        assert (
+            len(cels) == row_count
+        ), f'parsed row count ({len(cels)}) different from expected row count ({row_count})'
+
+    def _decorate(func):
+        @wraps(func)
+        def wrapped_function(self, *args, **kwargs):
+            return func(
+                self,
+                [CEL(id=i, **fields) for i, fields in enumerate(cels)],
+                *args,
+                **kwargs,
+            )
+
+        return wrapped_function
+
+    return _decorate
+
+
+def mock_dict(m: dict) -> MagicMock:
+    mock = MagicMock(m)
+    mock.__getitem__.side_effect = m.__getitem__
+    mock.__delitem__.side_effect = m.__delitem__
+    mock.__setitem__.side_effect = m.__setitem__
+
+    return mock
+
+
+def mock_user(
+    uuid: str,
+    tenant_uuid: str,
+    line_ids: list[str] | None = None,
+    mobile: str | None = None,
+    userfield: str | None = None,
+) -> dict[str, any]:
+    """
+    Create a mock user dictionary with the given parameters.
+
+    uuid (str): The UUID of the user.
+    tenant_uuid (str): The UUID of the tenant.
+    line_ids (list[str] | None, optional): The list of line IDs. Defaults to None.
+    mobile (str | None, optional): The mobile phone number. Defaults to None.
+    userfield (str | None, optional): The user field. Defaults to None.
+
+    Returns (dict[str, any]): The mock user dictionary.
+    """
```

Comment:
I'm sorry, but this docstring brings me no value, the parameters names and types are amply sufficient IMHO.

Reply:
Good point, will cleanup